lib/runAll.js

```diff
@@ -41,14 +41,14 @@ const shouldSkipApplyModifications = ctx => {
 
 const shouldSkipRevert = ctx => {
   // Should be skipped in case of unknown git errors
-  if (ctx.gitError && !ctx.gitApplyEmptyCommit && !ctx.gitApplyModificationsError) {
+  if (ctx.gitError && !ctx.gitApplyEmptyCommit && !ctx.gitRestoreUnstagedChanges) {
     return MESSAGES.GIT_ERROR
   }
 }
 
 const shouldSkipCleanup = ctx => {
   // Should be skipped in case of unknown git errors
-  if (ctx.gitError && !ctx.gitApplyEmptyCommit && !ctx.gitApplyModificationsError) {
+  if (ctx.gitError && !ctx.gitApplyEmptyCommit && !ctx.gitRestoreUnstagedChanges) {
```

Comment:
Maybe we should introduce state machines to code :)

Reply:
Maybe, but let's do that in a different PR.